src/features/contentCreator/addPluginPage.ts

```diff
@@ -0,0 +1,397 @@
+        </body>
+      </html>
+    `;
+  }
+
+  private _setWebviewMessageListener(webview: vscode.Webview) {
+    webview.onDidReceiveMessage(
+      async (message: any) => {
+        const command = message.command;
+        let payload;
+
+        switch (command) {
+          case "open-explorer": {
+            payload = message.payload;
+            const selectedUri = await this.openExplorerDialog(
+              payload.selectOption,
+            );
+            webview.postMessage({
+              command: "file-uri",
+              arguments: { selectedUri: selectedUri },
+            } as PostMessageEvent);
+            return;
+          }
+          case "init-create":
+            payload = message.payload as PluginFormInterface;
+            await this.runAddCommand(payload, webview);
+            return;
+
+          case "init-open-scaffolded-folder":
+            payload = message.payload;
+            await this.openFolderInWorkspace(
+              payload.projectUrl,
+              payload.pluginName,
+              payload.pluginType,
+            );
+            return;
+        }
+      },
+      undefined,
+      this._disposables,
+    );
+  }
+
+  private async getCreatorVersion(): Promise<string> {
+    const creatorVersion = (
+      await getBinDetail("ansible-creator", "--version")
+    ).toString();
+    console.log("ansible-creator version: ", creatorVersion);
+    return creatorVersion;
+  }
+
+  public async getCreatorCommand(
+    pluginName: string,
+    pluginType: string,
+    url: string,
+  ): Promise<string> {
+    let command = "";
+
+    command = `ansible-creator add plugin ${pluginType} ${pluginName} ${url} --no-ansi`;
+    return command;
+  }
+
+  public async openExplorerDialog(
+    selectOption: string,
+  ): Promise<string | undefined> {
+    const options: vscode.OpenDialogOptions = {
+      canSelectMany: false,
+      openLabel: "Select",
+      canSelectFiles: selectOption === "file",
+      canSelectFolders: selectOption === "folder",
+      defaultUri: vscode.Uri.parse(os.homedir()),
+    };
+
+    let selectedUri: string | undefined;
+    await vscode.window.showOpenDialog(options).then((fileUri) => {
+      if (fileUri && fileUri[0]) {
+        selectedUri = fileUri[0].fsPath;
+      }
+    });
+
+    return selectedUri;
+  }
+
+  public async runAddCommand(
+    payload: PluginFormInterface,
+    webView: vscode.Webview,
+  ) {
+    const { pluginName, pluginType, collectionPath, verbosity, isOverwritten } =
+      payload;
+
+    const destinationPathUrl = collectionPath
+      ? collectionPath
+      : `${os.homedir()}/.ansible/collections/ansible_collections`;
+
+    let ansibleCreatorAddCommand = await this.getCreatorCommand(
+      pluginName,
+      pluginType,
+      destinationPathUrl,
+    );
+
+    const creatorVersion = await this.getCreatorVersion();
+    if (isOverwritten) {
+      if (semver.gte(creatorVersion, ANSIBLE_CREATOR_VERSION_MIN)) {
```

Comment:
This is not needed as the add plugins feature got added recently to creator and we are not using the `--force` flag anymore. You can remove this check as well as the else condition where we add the `--force` flag. Instead, add the `--no-overwrite` parameter to the else condition.

Reply:
removed checks for `--force` and `ANSIBLE_CREATOR_VERSION_MIN`!